### ISSUE TITLE
When distributed training， why loss divided by "num_gpus"?

### ISSUE THREAD

**Issue by Leslie0115** [tianzhi0549/FCOS#275]
Hi, I found that loss is divided by "num_pos_avg_per_gpu", and "num_pos_avg_per_gpu = total_num_pos / num_gpus", here I have a question about why it is divided by "num_gpus". 
I checked the pytorch manual about distributed training, I think each GPU does its own complete loss calculation in its own process, for example, batchsize is 8 and  there are two GPUs, I think that 4 inputs for GPU0 and 4 inputs for GPU1, each GPU calls "loss.py" separately in its own process,  and each "loss.py" calls "sigmoid_focal_loss.py" with 4 inputs. 
But in your code, it seems that a master process calls "loss.py", and "loss.py" calls "sigmoid_focal_loss.py" with 8 inputs, and then the focal_loss returned to "loss.py" is divided by "num_gpus", so I'm confused  which steps are completed in their respective processes and which are completed in a master process? Is there something wrong with my understanding of distributed?
I'm not sure if I describe my problem clearly, if you could understand my question,  I will be very grateful if you can tell.

**Comment by tianzhi0549** [comment]
@Leslie0115 First, in our code, each process computes its own losses as in standard DDP training. We divide `num_gpus`  because we hope each GPU has the same number of positive samples, thus the number of positive samples is synchronized before this division.

**Comment by Leslie0115** [comment]
@tianzhi0549 I fully understand the meaning of your doing this. I was confused before because I have never seen the definition of the "reduce_sum()", I always thought it was the function that in pytorch. Thanks for your answer!
